src/pages/NIHicWebform.js

```diff
@@ -0,0 +1,2795 @@
+import { Component } from 'react';
+import { RadioButton } from '../components/RadioButton';
+import { a, br, div, fieldset, form, h, h3, hr, input, label, span, textarea } from 'react-hyperscript-helpers';
+import { Link } from 'react-router-dom';
+import Select from 'react-select';
+import ReactTooltip from 'react-tooltip';
+import { Alert } from '../components/Alert';
+import { Notification } from '../components/Notification';
+import { PageHeading } from '../components/PageHeading';
+import {DAR, DataSet} from '../libs/ajax';
+import { searchOntology } from '../libs/ontologyService';
+import * as fp from 'lodash/fp';
+import AsyncSelect from 'react-select/async';
+
+class NIHICWebform extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dacList: [],
+      selectedDac: {},
+      allDatasets: '',
+      allDatasetNames: [],
+      updateDataset: {},
+      nihValid: false,
+      disableOkBtn: false,
+      showValidationMessages: false,
+      showModal: false,
+      showDialogSubmit: false,
+      formData: {
+        methods: '',
+        genetic: '',
+        publication: '',
+        collaboration: '',
+        ethics: '',
+        geographic: '',
+        moratorium: '',
+        populationMigration: '',
+        forProfit: '',
+        hmb: false,
+        poa: false,
+        diseases: false,
+        ontologies: [],
+        other: false,
+        otherText: '',
+        generalUse: false
+      },
+      datasetData: {
+        datasetName: '',
+        researcher: '',
+        collectionId: '',
+        pi: '',
+        datasetRepoUrl: '',
+        dataType: '',
+        species: '',
+        phenotype: '',
+        nrParticipants: '',
+        description: '',
+        dac: '',
+        consentId: '',
+        publicAccess: false,
+      },
+      problemSavingRequest: false,
+      problemLoadingUpdateDataset: false,
+      submissionSuccess: false,
+      errorMessage: ''
+    };
+
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+  };
+
+  onNihStatusUpdate = (nihValid) => {
+    if (this.state.nihValid !== nihValid) {
+      this.setState(prev => {
+        prev.nihValid = nihValid;
+        return prev;
+      });
+    }
+  };
+
+
+  async init() {
+    const { datasetId } = this.props.match.params;
+    let updateDataset = {};
+    // update dataset case
+    if (!fp.isNil(datasetId)) {
+      DataSet.getDataSetsByDatasetId(datasetId).then(data => {
```

Comment:
If you're not logged in (and this is an open demo page) then this call will fail